File: src/main/java/com/wgzhao/addax/admin/config/HiveJdbcCondition.java
package com.wgzhao.addax.admin.config;

import org.springframework.context.annotation.Condition;
import org.springframework.context.annotation.ConditionContext;
import org.springframework.core.type.AnnotatedTypeMetadata;

public class HiveJdbcCondition implements Condition
{
    @Override
    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata)
    {
        String impl = context.getEnvironment().getProperty("target.service.impl");
        return "jdbc".equalsIgnoreCase(impl);
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/config/HiveJdbcCondition.kt
package com.wgzhao.addax.admin.config

import org.springframework.context.annotation.Condition
import org.springframework.context.annotation.ConditionContext
import org.springframework.core.type.AnnotatedTypeMetadata

class HiveJdbcCondition : Condition {
    override fun matches(context: ConditionContext, metadata: AnnotatedTypeMetadata): Boolean {
        val impl = context.getEnvironment().getProperty("target.service.impl")
        return "jdbc".equals(impl, ignoreCase = true)
    }
}

